baremetal_network_provisioning/tests/unit/ml2/test_hp_network_provisioning_driver.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2015 OpenStack Foundation
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+from baremetal_network_provisioning.common import constants as hp_const
+from baremetal_network_provisioning.common import exceptions as hp_exec
+from baremetal_network_provisioning.db import bm_nw_provision_db as db
+from baremetal_network_provisioning.db import bm_nw_provision_models as models
+from baremetal_network_provisioning.ml2 import hp_network_provisioning_driver as driver
+
+import mock
+from oslo_config import cfg
+import requests
+
+from neutron.tests import base
+
+CONF = cfg.CONF
+
+
+class TestHPNetworkProvisioningDriver(base.BaseTestCase):
+
+    def setUp(self):
+        super(TestHPNetworkProvisioningDriver, self).setUp()
+        CONF.set_override('base_url', 'fake_url', 'default')
+        self.driver = driver.HPNetworkProvisioningDriver()
+
+    def test_create_port_with_200_ok(self):
+        """test_create_port_with_valid_device."""
+        port_dict = self._get_port_payload()
+        res_200 = FakeResponse(200)
+        with mock.patch.object(self.driver,
+                               '_do_request',
+                               return_value=res_200):
+            with mock.patch.object(db, 'add_hp_switch_port',
+                                   return_value=None):
+                with mock.patch.object(db, 'add_hp_ironic_switch_port_mapping',
+                                       return_value=None):
+                    self.driver.create_port(port_dict)
+
+    def test_create_port_with_connection_failed(self):
+        """test_create_port_with_connection_failed."""
```

Comment:
docstring should be some description about the method.
"""" Test create_port_with_connection_failed method."""